### DISCUSSION TITLE
Is the WebXR Device API ready to be used in production?

### DISCUSSION THREAD

**Discussion by hd-esteban** [google/model-viewer#2069] [Q&A]
I've been checking the documentation related with the WebXR Device API and it seems that Google already released some working examples and also I noticed you have some examples in your documentation using it.

https://modelviewer.dev/examples/augmentedreality/index.html#demo-container-2

But when I tested them on my mobile device (huawei mate 20 lite) the 3d models are not getting displayed properly and also the viewer that uses WebXR Device API seems to be really slow. 

So I would like to know the current status of the webxr api, if this is stable or it is still a work in progress that is not ready to be used on production environments.

**Comment by elalish** [comment]
Well, it depends on what your bar is. WebXR is in stable Chrome and it's our default AR mode in model-viewer, so yes, we stand behind it. However, AR in general is a bit bleeding edge, not supported on all devices, and too slow on lower-end GPUs. If you find a specific shortcoming between WebXR relative to SceneViewer (Android's native AR mode), feel free to file an issue on it and ensure you fill in the entire template. 

Chrome is actively chasing down device-specific rendering issues and working to improve performance, so WebXR is continuing to improve.

**Reply by hd-esteban** [reply]
Thank you for your quick reply @elalish, I've another question regarding this, where is the official documentation about how to implement the Webxr API? I would like to build a simple viewer without all the functionality that Model viewer offers so I think it is better for our purposes to build it from the scratch.

**Reply by elalish** [reply]
We had a rule in grad school that if someone could find the answer to your question with less than a minute of google searching, you owed them a dollar: https://developer.mozilla.org/en-US/docs/Web/API/WebXR_Device_API 😁

**Reply by hd-esteban** [reply]
haha, thanks, actually I saw that doc yesterday, but I had some doubts if it was the most recent and complete documentation or if it is what is shown in this link: https://github.com/immersive-web/webxr but yes, I owe you a dollar.

**Reply by hd-esteban** [reply]
I already reported the bugs we found with your team, it not seems to be something related with device or the software versions because the Scene Viewer implementations doesn't have problems while rendering products.
